src/rgw/rgw_rados.cc

```diff
@@ -3384,6 +3444,18 @@ class RGWRadosPutObj : public RGWHTTPStreamRWRequest::ReceiveCB
   uint64_t get_data_len() {
     return data_len;
   }
+
+  string get_calculated_etag() {
+    if (obj_type == OBJ_TYPE_ATOMIC) {
+      etag_verifier_atomic->calculate_etag();
```

Comment:
can just do if (etag_verifier_atomic) ... else if (etag_verifier_mpu) without the extra enum

Reply:
But I see that there's a third case as well where we have to return the empty string. I used to `OBJ_TYPE_UNINIT` to track the same.